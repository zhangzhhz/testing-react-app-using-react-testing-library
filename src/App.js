import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import logo from './logo.svg';
import './App.css';
import TestRouter from './TestRouter';

function App() {
  return (
    <div className="App">
      <Router history={createMemoryHistory()}>
        <TestRouter/>
      </Router>
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
