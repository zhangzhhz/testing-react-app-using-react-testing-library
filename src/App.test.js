import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('should take a snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
