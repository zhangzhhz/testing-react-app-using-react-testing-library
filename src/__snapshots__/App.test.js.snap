// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      data-testid="navbar"
    >
      <a
        data-testid="home-link"
        href="/"
      >
        Home
      </a>
      <a
        data-testid="about-link"
        href="/about"
      >
        About
      </a>
      <a
        data-testid="contact-link"
        href="/contact/John Doe"
      >
        Contact
      </a>
    </nav>
    <h1
      data-testid="home-page"
    >
      Home page
    </h1>
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
